charts/s1-agent/templates/common/configmaps.yaml

```diff
@@ -30,3 +30,19 @@ data:
         configMap:
           name: {{ include "agent.fullname" . }}
 {{ end }}
+
+---
+
+{{ if or (not .Values.configuration.env.injection.enabled) (eq (include "serverlessOnlyMode" .) "false") }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ include "agent.fullname" . }}-config
```

Comment:
Google are supposed to whitelist specific names for config maps AFAIU, maybe we prefer to have something hardcoded here that cannot dynamically change?

Reply:
We generate a flattened yaml for gke autopilot, so eventually it will contain the default hardcoded settings, i.e. s1-agent-config. It will require autopilot customers to use the defaults when installing the agent.
We have other cases like this and I think we should allow this flexibility for non-gke-autopilot customers